pkg/version/version.go

```diff
@@ -92,5 +91,42 @@ func GetLatestVersion(uuid string, source string) (string, error) {
 		return "", err
 	}
 
-	return versionRsp["version"], nil
+	channel := parseChannel(Version)
+	if channel == "" {
+		channel = "version"
+	}
+
+	version, ok := versionRsp[channel]
+	if !ok {
+		return "", fmt.Errorf("Unsupported version channel: %s", channel)
+	}
+
+	return version, nil
+}
+
+func parseVersion(version string) string {
+	if parts := strings.SplitN(version, "-", 2); len(parts) == 2 {
+		return parts[1]
+	}
+	return version
+}
+
+func parseChannel(version string) string {
```

Comment:
similar to comment earlier, should `parseVersion` and `parseChannel` return errors if no channel is found in the version?